conduct/wallets/base.py

```diff
@@ -1,30 +1,75 @@
+import requests
+
 from abc import ABC, abstractmethod
-from typing import Optional
+from http import HTTPStatus
+from requests import Response
+
+from conduct.types import Invoice, MilliSatoshi
 
 
 class Wallet(ABC):
     """Abstract class."""
 
     @abstractmethod
-    async def get_info(self):
+    def get_info(self):
         pass
 
     @abstractmethod
-    async def get_balance(self):
+    def get_balance(self) -> MilliSatoshi:
         pass
 
     @abstractmethod
-    async def create_invoice(self, *, amount: Optional[int], memo: Optional[str]):
+    def create_invoice(
+        self, *, amount: int, description: str = "", expiry: int = 3600
+    ) -> Invoice:
         pass
 
     @abstractmethod
-    async def pay_invoice(self, *, payment_request: str):
+    def pay_invoice(self, *, payment_request: str):
         pass
 
     @abstractmethod
-    async def get_invoice_status(self, *, invoice_id: str):
+    def get_invoice_status(self, *, txid: str):
```

Comment:
missing return type?

Reply:
I would like to discuss the returns: this is not decided yet. It could be similar to what we now have in LNbits: a response that includes data + errors or manage errors in a different way.